clippy_lints/src/large_enum_variant.rs

```diff
@@ -113,23 +114,27 @@ impl<'tcx> LateLintPass<'tcx> for LargeEnumVariant {
                                 def.variants[(second.1).0].span,
                                 &format!("and the second-largest variant is {} bytes:", second.0),
                             );
-                            if variant.fields.len() == 1 {
-                                let span = match def.variants[i].data {
-                                    VariantData::Struct(fields, ..) | VariantData::Tuple(fields, ..) => {
-                                        fields[0].ty.span
-                                    },
-                                    VariantData::Unit(..) => unreachable!(),
-                                };
-                                if let Some(snip) = snippet_opt(cx, span) {
-                                    diag.span_suggestion(
-                                        span,
-                                        help_text,
-                                        format!("Box<{}>", snip),
-                                        Applicability::MaybeIncorrect,
-                                    );
-                                    return;
+                            let mut sugg: Vec<(Span, String)> = vec![];
+                            let fields = match def.variants[i].data {
+                                VariantData::Struct(fields, ..) | VariantData::Tuple(fields, ..) => fields,
+                                VariantData::Unit(..) => unreachable!(),
+                            };
+                            let length = variant.fields.len() as u64;
+                            for (ind, val) in variant.fields.iter().enumerate() {
+                                let ty = cx.tcx.type_of(val.did);
+                                if let Some(size_info) = cx.layout_of(ty).ok().map(|l| l.size.bytes()) {
+                                    if size_info >= self.maximum_size_difference_allowed / length {
```

Comment:
This isn't quite right. We should Box as many fields as needed to bring the size difference below what is allowed.

Reply:
> This isn't quite right. We should Box as many fields as needed to bring the size difference below what is allowed.

Done. Thanks for your patient review.